lib/src/chart/gauge_chart/gauge_chart_data.dart

```diff
@@ -0,0 +1,268 @@
+import 'dart:ui';
+
+import 'package:equatable/equatable.dart';
+import 'package:fl_chart/fl_chart.dart';
+import 'package:fl_chart/src/utils/lerp.dart';
+import 'package:flutter/widgets.dart';
+
+class ColoredTick with EquatableMixin {
+  const ColoredTick(this.position, this.color);
+  final double position;
+  final Color color;
+
+  @override
+  List<Object?> get props => [position, color];
+}
+
+mixin ColoredTicksGenerator {
+  Iterable<ColoredTick> getColoredTicks();
+}
+
+@immutable
+abstract class GaugeColor {
+  Color getColor(double value);
+
+  static GaugeColor lerp(GaugeColor a, GaugeColor b, double t) {
+    return _LerpGaugeColor(a, b, t);
+  }
+}
+
+class _LerpGaugeColor implements GaugeColor, ColoredTicksGenerator {
+  _LerpGaugeColor(this.a, this.b, this.t);
+  final GaugeColor a;
+  final GaugeColor b;
+  final double t;
+
+  @override
+  Color getColor(double value) {
+    return Color.lerp(a.getColor(value), b.getColor(value), t)!;
+  }
+
+  @override
+  Iterable<ColoredTick> getColoredTicks() sync* {
+    if (a is ColoredTicksGenerator) {
+      for (final tick in (a as ColoredTicksGenerator).getColoredTicks()) {
+        yield ColoredTick(tick.position, tick.color.withOpacity(1 - t));
+      }
+    }
+    if (b is ColoredTicksGenerator) {
+      for (final tick in (b as ColoredTicksGenerator).getColoredTicks()) {
+        yield ColoredTick(tick.position, tick.color.withOpacity(t));
+      }
+    }
+  }
+}
+
+@immutable
+class SimpleGaugeColor with EquatableMixin implements GaugeColor {
+  const SimpleGaugeColor({required this.color});
+  final Color color;
+
+  @override
+  Color getColor(double value) => color;
+
+  @override
+  List<Object?> get props => [color];
+}
+
+class VariableGaugeColor
```

Comment:
This is a confusing approach in this project. Because we didn't use this approach on the other charts.
I would prefer to have a callback to get the progress and return a color (what we do in other charts).
Or a list of colors if you want to handle the gradient.
By the way, I can add these kinds of improvements if you allow me

Reply:
The idea is to be able to display some colored ticks at the angle the color changes. 
If you work with a callback, you cannot get this angle precisely and moreover your will need to call it a lot of times with values between 0 and 1 to scan when the color changes.